### PR TITLE
code changes for cephadm-ansible

### PR DESCRIPTION
**Pull request by sunilkumarn417**
Signed-off-by: sunilkumarn417 <sunnagar@redhat.com>

- Added cephadm-ansible support in automation.
- updated build to build upgrade test suite with cephadm-ansible playbooks.
   - cephadm-preflight
   - cephadm-purge-cluster
   - cephadm-clients

**Test results:**
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-VNONDM/

### CODE REVIEW

**Review by Ameenasuhani — COMMENTED** [review]
can the fsid be a default?

File: suites/pacific/upgrades/tier-1_upgrade_cephadm.yaml [hunk 1]
```diff
@@ -154,6 +159,15 @@ tests:
...
       abort-on-fail: true
+#  - test:
+#      name: Cephadm-ansible client playbook
+#      desc: Configure cephadm client pre-req using cephadm-clients.yml playbook
+#      polarion-id:
+#      module: test_cephadm_ansible.py
+#      config:
+#        playbook: cephadm-clients.yml
+#        extra-vars:
+#          fsid: f64f341c-655d-11eb-8778-fa163e914bcc
```
Comment (Ameenasuhani) [inline suites/pacific/upgrades/tier-1_upgrade_cephadm.yaml:170]: Any reason why it is not included in suite?

**Comment by sunilkumarn417** [comment]
> can the fsid be a default?

yes, it can be default or it can be pre-fixed. For now we are using dedicated id here until we brake out further more scenarios in cephadm-ansible.

File: suites/pacific/upgrades/tier-1_upgrade_cephadm.yaml (hunk 1 above)
Comment (sunilkumarn417) [reply suites/pacific/upgrades/tier-1_upgrade_cephadm.yaml:170]: @Ameenasuhani need some insights here.

Comment (Ameenasuhani) [reply suites/pacific/upgrades/tier-1_upgrade_cephadm.yaml:170]: @sunilkumarn417 are we not executing cephadm-clients playbook here?

Comment (Ameenasuhani) [reply suites/pacific/upgrades/tier-1_upgrade_cephadm.yaml:170]: https://github.com/ceph/cephadm-ansible/blob/devel/cephadm-clients.yml#L1-L24 does this info help?

Comment (psathyan) [reply suites/pacific/upgrades/tier-1_upgrade_cephadm.yaml:170]: We could either reduce the scope of this PR to remove those tests or have them without commenting it. The pipeline script could be added once the BZ is resolved.

File: suites/pacific/upgrades/tier-1_upgrade_cephadm.yaml [hunk 2]
```diff
@@ -37,7 +37,7 @@ tests:
       name: Cluster deployment
       desc: Cluster deployment with latest released version of 5.x
       module: test_cephadm.py
-      polarion-id: CEPH-83573791
+      polarion-id: CEPH-83573791,CEPH-83574413
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply suites/pacific/upgrades/tier-1_upgrade_cephadm.yaml:40]: I think this need split of repository configuration and deployment as different test cases.

File: ceph/ceph_admin/cephadm_ansible.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        def entries(lst, group_name=None):
+            lst.insert(0, f"\n[{group_name}]" if group_name else "")
+            return "\n".join(lst)
+
+        inventory = entries(others)
+        inventory += entries(admins, "admin")
+        inventory += entries(clients, "clients")
+
+        inv_file = self.admin.remote_file(
+            sudo=True, file_name=self.inventory, file_mode="w"
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply ceph/ceph_admin/cephadm_ansible.py:84]: `/usr/share/cephadm-ansible` needs root permission to manage files and directories.
Currently it's been already used for ceph-ansible and working in IBM-c. https://github.com/red-hat-storage/cephci/blob/master/ceph/ceph.py#L2152-L2158

File: suites/pacific/upgrades/tier-1_upgrade_cephadm.yaml (hunk 1 above)
Comment (sunilkumarn417) [reply suites/pacific/upgrades/tier-1_upgrade_cephadm.yaml:170]: working on it. let me check this out once again.
Thanks @psathyan @Ameenasuhani .

Comment (sunilkumarn417) [reply suites/pacific/upgrades/tier-1_upgrade_cephadm.yaml:170]: good that, we found a issue in cephadm-clients playbook.

Comment (sunilkumarn417) [reply suites/pacific/upgrades/tier-1_upgrade_cephadm.yaml]: However, code is in and we can enable this as soon as BZ is fixed. https://bugzilla.redhat.com/show_bug.cgi?id=2039413

File: ceph/ceph_admin/cephadm_ansible.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (psathyan) [reply ceph/ceph_admin/cephadm_ansible.py:84]: Okay.

**Review by psathyan — APPROVED** [review]
LGTM

**Pull request merged**